src/bspwm.c

```diff
@@ -193,6 +204,25 @@ int main(int argc, char *argv[])
 		}
 	}
 
+	char restart_state_path[MAXLEN];
+	char *rsp = getenv(RESTART_STATE_ENV_VAR);
+	if (rsp != NULL) {
+		snprintf(restart_state_path, sizeof(restart_state_path), "%s", rsp);
+	} else {
+		char *host = NULL;
+		int dn = 0, sn = 0;
+		if (xcb_parse_display(NULL, &host, &dn, &sn) != 0) {
+			snprintf(restart_state_path, sizeof(restart_state_path), RESTART_STATE_PATH_TPL, host, dn, sn);
+		}
+		free(host);
+	}
+
+	if (restart) {
+		FILE *f = fopen(restart_state_path, "w");
```

Comment:
Maybe if `RESTART_STATE_ENV_VAR` is not set, the temporary file should be cleaned up?